state/core_access.go

```diff
@@ -223,14 +254,17 @@ func (ca *CoreAccessor) SubmitPayForBlob(
 
 	var lastErr error
 	for attempt := 0; attempt < maxRetries; attempt++ {
+		options := []apptypes.TxBuilderOption{apptypes.SetGasLimit(gasLim), withFee(fee)}
```

Comment:
maybe you can define the options a bit higher up and just do the append to opts inline instead of checking `feeGrant != nil`, 

a bit cleaner

Reply:
gasLimit and fee options could be recalculated, so we will need to clear the slice(besides grantFee opt) on every attempt. That's why I decided to create a new slice